compose/progress_stream.py

```diff
@@ -14,8 +14,14 @@ def stream_output(output, stream):
 
     for event in utils.json_stream(output):
         all_events.append(event)
+        is_progress_event = 'progress' in event or 'progressDetail' in event
 
-        if 'progress' in event or 'progressDetail' in event:
+        if not is_progress_event:
+            print_output_event(event, stream, is_terminal)
+            stream.flush()
+
+        # if it's a progress event and we have a terminal, then display the progress bars
+        elif is_terminal:
```

Comment:
I took another look at this with @aanand 

I think by making this a couple of guard statements it would be slightly easier to understand. How about this:

```python
if not is_progress_event:
    ...
   continue

if not is_terminal:
    continue

...
```

That way the rest of the logic is outside of the if/elif block